prioritization_fees/src/account_priofees.rs

```diff
@@ -0,0 +1,220 @@
+use std::{
+    collections::{BTreeMap, HashMap},
+    sync::{atomic::AtomicU64, Arc},
+};
+
+use dashmap::{mapref::multiple::RefMutMulti, DashMap};
+use itertools::Itertools;
+use solana_lite_rpc_core::structures::produced_block::ProducedBlock;
+use solana_sdk::pubkey::Pubkey;
+
+use crate::{
+    prioritization_fee_data::{BlockPrioData, PrioFeesData},
+    rpc_data::AccountPrioFeesStats,
+};
+
+pub struct AccountPrio {
+    pub stats_by_slot: BTreeMap<u64, BlockPrioData>,
+}
+
+#[derive(Clone)]
+pub struct AccountPrioStore {
+    pub account_by_prio_fees_all: Arc<DashMap<Pubkey, AccountPrio>>,
+    pub account_by_prio_fees_writeonly: Arc<DashMap<Pubkey, AccountPrio>>,
+    pub number_of_slots_to_save: usize,
+    pub last_slot: Arc<AtomicU64>,
+}
+
+impl AccountPrioStore {
+    pub fn new(number_of_slots_to_save: usize) -> Self {
+        Self {
+            account_by_prio_fees_all: Arc::new(DashMap::new()),
+            account_by_prio_fees_writeonly: Arc::new(DashMap::new()),
+            number_of_slots_to_save,
+            last_slot: Arc::new(AtomicU64::new(0)),
+        }
+    }
+
+    pub fn update(&self, produced_block: &ProducedBlock) {
+        // sort by ascending order
+        let transactions = produced_block
+            .transactions
+            .iter()
+            .filter(|x| !x.is_vote)
+            .sorted_by(|a, b| a.prioritization_fees.cmp(&b.prioritization_fees))
+            .rev();
+        // accounts
+        let mut accounts_by_prioritization_write: HashMap<Pubkey, Vec<PrioFeesData>> =
+            HashMap::new();
+        let mut accounts_by_prioritization_read_write: HashMap<Pubkey, Vec<PrioFeesData>> =
+            HashMap::new();
+        for transaction in transactions {
+            let value = PrioFeesData {
+                priority: transaction.prioritization_fees.unwrap_or_default(),
+                cu_consumed: transaction.cu_consumed.unwrap_or_default(),
+            };
+            for write_lock in &transaction.writable_accounts {
+                match accounts_by_prioritization_write.get_mut(write_lock) {
+                    Some(acc_vec) => {
+                        acc_vec.push(value);
+                    }
+                    None => {
+                        accounts_by_prioritization_write.insert(*write_lock, vec![value]);
+                    }
+                }
+
+                match accounts_by_prioritization_read_write.get_mut(write_lock) {
+                    Some(acc_vec) => {
+                        acc_vec.push(value);
+                    }
+                    None => {
+                        accounts_by_prioritization_read_write.insert(*write_lock, vec![value]);
+                    }
+                }
+            }
+
+            for readlock in &transaction.readable_accounts {
+                match accounts_by_prioritization_read_write.get_mut(readlock) {
+                    Some(acc_vec) => {
+                        acc_vec.push(value);
```

Comment:
there will be two items added to vec if `for write_lock` and `for readlock` is matching. is that correct?

Reply:
Yeah one is only write_lock. other is both write_lock and read lock.